examples/evaluation/evaluate_on_binary_classifier.py

```diff
@@ -0,0 +1,31 @@
+import xgboost
+import shap
+import mlflow
+from sklearn.model_selection import train_test_split
+
+# train XGBoost model
+X, y = shap.datasets.adult()
+
+num_examples = len(X)
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+
+model = xgboost.XGBClassifier().fit(X_train, y_train)
+
+eval_data = X_test
+eval_data["label"] = y_test
+
+with mlflow.start_run() as run:
+    mlflow.sklearn.log_model(model, "model")
+    model_uri = mlflow.get_artifact_uri("model")
+    result = mlflow.evaluate(
+        model_uri,
+        eval_data,
+        targets="label",
+        model_type="classifier",
+        dataset_name="adult",
+        evaluators=["default"],
+    )
+
+print(f"metrics:\n{result.metrics}")
+print(f"artifacts:\n{result.artifacts}")
```

Comment:
This line's output looks like:

```
{'roc_curve_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d98244950>, 'precision_recall_curve_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d982890d0>, 'lift_curve_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d985bc250>, 'confusion_matrix': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d9826bc10>, 'shap_beeswarm_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d98123fd0>, 'shap_summary_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d2c18fd50>, 'shap_feature_importance_plot': <mlflow.models.evaluation.artifacts.ImageEvaluationArtifact object at 0x7f5d96eb1990>}
```

We might want to implement `__str__` for `EvaluationArtifact` for better string representation (e.g. `<EvaluationArtifact file_name.png>`).

cc @dbczumar

Reply:
We can do this in follow-up updates.

Reply:
I add a `__repr__` for evaluation artifact , format is like `ImageEvaluationArtifact(uri='...')`